engine/jobmaster/dm/task_manager.go

```diff
@@ -44,15 +46,26 @@ type TaskManager struct {
 	// tasks record the runtime task status
 	// taskID -> TaskStatus
 	tasks sync.Map
+
+	gaugeVec   *prometheus.GaugeVec
+	stageGauge map[string]prometheus.Gauge
 }
 
 // NewTaskManager creates a new TaskManager instance
-func NewTaskManager(initTaskStatus []runtime.TaskStatus, jobStore *metadata.JobStore, messageAgent dmpkg.MessageAgent, pLogger *zap.Logger) *TaskManager {
+func NewTaskManager(initTaskStatus []runtime.TaskStatus, jobStore *metadata.JobStore, messageAgent dmpkg.MessageAgent, pLogger *zap.Logger, metricFactory promutil.Factory) *TaskManager {
 	taskManager := &TaskManager{
 		DefaultTicker: ticker.NewDefaultTicker(taskNormalInterval, taskErrorInterval),
 		jobStore:      jobStore,
 		logger:        pLogger.With(zap.String("component", "task_manager")),
 		messageAgent:  messageAgent,
+		gaugeVec: metricFactory.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace: "job_master",
```

Comment:
Does the monitor name here need to be compatible with `dm_worker_task_state`? This makes it easier for the dm and engine to share the monitor panel.

Reply:
It is ok for me.